### DISCUSSION TITLE
How use non-unique field on deletion/updation?

### DISCUSSION THREAD

**Discussion by cawa-93** [prisma/prisma#14012] [Q&A] [answered]
Hi. My schema:
```prisma
model User {
  id         Int         @id @default(autoincrement())
  templates  Template[]
}

model Template {
  id                     Int             @id @default(autoincrement())
  ownerId                Int
  owner                  User            @relation(fields: [ownerId], references: [id], onDelete: Cascade)
}
```

Next, I create an API to delete templates for the transmitted ID. However, it is important that each authorized user can only delete their own templates. Usually I would generate the following query:

```sql
DELETE FROM "Template" WHERE id = <input.id> and "ownerId" = <context.user.id>
```

However, the prisma does not allow me to generate such a query, throwing type error:
```ts
db.template.delete({
  where: {
    id: input.id,
    ownerId: ctx.user.id, // Type '{ id: number; ownerId: number; }' is not assignable to type 'TemplateWhereUniqueInput'. Object literal may only specify known properties, and 'ownerId' does not exist in type 'TemplateWhereUniqueInput'.
  }
})
```
```ts
// .prisma/client/index.d.ts
export type TemplateWhereUniqueInput = {
  id?: number
}
```

I found a way to do what I want by deleting the templates as requested by the user update:
```ts
db.user.update({
  where: {
    id: ctx.user.id
  },
  data: {
    templates: {
      delete: {
        id: input.id
      }
    }
  }
})
```
However, it is difficult to read and also makes the code verbose. 

Is there a way to make the same easier?

**Comment by austincrim** [comment]
Hey @cawa-93 👋,

Try adding a unique index to `Template` that consists of `id` and `ownerId`.
```prisma
model User {
  id         Int         @id @default(autoincrement())
  templates  Template[]
}

model Template {
  id                     Int             @id @default(autoincrement())
  ownerId                Int
  owner                  User            @relation(fields: [ownerId], references: [id], onDelete: Cascade)
  
  @@unique([id, ownerId])
}
```

Then you can update like so:
```ts
await prisma.template.update({
  where: {
    id_ownerId: {
      id: input.id,
      ownerId: ctx.user.id
    }
  },
  data: {...}
})
```

**Comment by npmrun** [comment]
any help?

I dont think owerId a unique field。

Have other advice today?

**Reply by janpio** [reply, marked as answer]
Does this help?
https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#enable-the-ability-to-filter-on-non-unique-fields-with-userwhereuniqueinput
https://github.com/prisma/prisma/issues/15837

**Reply by npmrun** [reply]
Thanks a lot  !!!
This will solve my problem.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
